src/analysis/SmartDetectorRunnerChildProcess/SmartDetectorRunnerMain.cs

```diff
@@ -67,12 +64,22 @@ public static int Main(string[] args)
         /// </summary>
         /// <param name="request">The Smart Detector request</param>
         /// <param name="cancellationToken">The cancellation token</param>
-        /// <returns>A <see cref="Task{TResult}"/>, returning the alerts presentations generated by the Smart Detector</returns>
-        private static async Task<List<ContractsAlert>> RunSmartDetectorAsync(SmartDetectorAnalysisRequest request, CancellationToken cancellationToken)
+        /// <returns>A <see cref="Task{TResult}"/>, returning the result of the call to the runner.</returns>
+        private static async Task<object> RunSmartDetectorAsync(SmartDetectorRunnerChildProcessInput request, CancellationToken cancellationToken)
         {
             ISmartDetectorRunner smartDetectorRunner = container.Resolve<ISmartDetectorRunner>();
             bool shouldDetectorTrace = bool.Parse(ConfigurationReader.ReadConfig("ShouldDetectorTrace", required: true));
-            return await smartDetectorRunner.RunAsync(request, shouldDetectorTrace, cancellationToken);
+
+            if (request.AnalysisRequest != null)
+            {
+                return await smartDetectorRunner.AnalyzeAsync(request.AnalysisRequest, shouldDetectorTrace, cancellationToken);
+            }
+            else if (request.AutomaticResolutionCheckRequest != null)
```

Comment:
>AutomaticResolution [](start = 29, length = 19)

I wonder if "automatic resolution" is the rights name.
From the point of view of the person that uses the detector it's indeed automatic. But from the point of view of detector developer nothing happens automatic here, she needs to write the resolution logic.

Reply:
a. I do think it's the right functional name, as it distinguishes between automatic (== code) and manual (== user) resolution
b. feel free to come up with a suggestion

---
In reply to: [234985079](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r234985079) [](ancestors = 234985079)